Separate Technologies chip into Technology and Language chips

luisitocanlas opened the pull request:
Fixes #4648

### What changes did you make?
  - Refactored current-projects.js so that "Language" is rendered on the chip when a language is selected and "Technology" for a selected technology.
  - Added a helper function `function addToDropDownFilterComponents` to add filter components to already existing filter category
  - Added `if else` block to add languages inside the technologies filter-item group
  - Separated technologies and languages at lines 257 and 258
  - Refactored `function updateCategoryCounter`

### Why did you make the changes (we will use this info to test)?
  - Per issue #4648, we need to separate technologies from languages so that each technology or language will have their own "Technology" and "Language" chip label, respectively.
  - For `if else` block, I needed to put the `languages` under the same filter item category as `technologies` and have them sorted.
  - For lines 257 and 258, I separated them because `languages` are being tied up to the `technologies` filter name.
  - For `function updateCategoryCounter`, the counter will only add up either `technologies` or `languages`. For example, when you select 3 `technologies` the counter will show `(3)` but when you select a `language` while having those `techonologies` selected, it will show `(1)`.

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

![image](https://github.com/hackforla/website/assets/78521256/00bf5331-bc1e-47e3-a495-8d79c3e44d5a)

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
![image](https://github.com/hackforla/website/assets/78521256/b3c405bd-f3fb-4e04-a934-44e94b012b12)

</details>

### Noted difference between the live website and the refactored current-projects.js when multiple filters are selected. Please advise.
 - live website will show `languages` OR `technologies`
 - refactored current-projects.js will show `languages` AND `technologies`

luisitocanlas committed:
Fixed the technology and language filters

colin-macrae commented:
Availability: 6 hrs during 3/18-22
ETA: 3/22

danvgar commented:
Happy to review this. Will provide a response within the next 48 hours!

irais-valenzuela commented:
Review ETA: 3/20/24
Availability:  3/20/24

irais-valenzuela reviewed (dismissed):
Hi @luisitocanlas, great job on the issue! Your branches are correct, there's a linked issue, and you successfully wrote the code to separate the technologies and languages on the chip labels. Thanks for all the comments and detailed documentation it was super helpful!

danvgar reviewed (commented):
@luisitocanlas I've reviewed this pretty thoroughly and think it looks good. I really appreciate your thorough PR description and the comments throughout.

My only real question is related to L494, with the comment `//might need to add languages here, for issue #4648`. Could you clarify what this comment's purpose is?

luisitocanlas commented:
> @luisitocanlas I've reviewed this pretty thoroughly and think it looks good. I really appreciate your thorough PR description and the comments throughout.
> 
> My only real question is related to L494, with the comment `//might need to add languages here, for issue #4648`. Could you clarify what this comment's purpose is?

@danvgar I figured since we're separating the technologies and languages, I added 'language' at the end of the searchAreas. But I didn't really see any effect when I added that. Do you want me to keep it or revert it back to what it was before? Or any other recommendations for that?

Thank you!

danvgar commented:
> > @luisitocanlas I've reviewed this pretty thoroughly and think it looks good. I really appreciate your thorough PR description and the comments throughout.
> > My only real question is related to L494, with the comment `//might need to add languages here, for issue #4648`. Could you clarify what this comment's purpose is?
> 
> @danvgar I figured since we're separating the technologies and languages, I added 'language' at the end of the searchAreas. But I didn't really see any effect when I added that. Do you want me to keep it or revert it back to what it was before? Or any other recommendations for that?
> 
> Thank you!

You're all good! I was mainly confused by the wording of it. My opinion is comments should be as declarative as possible, with any discrepancy or debate left for other forms of documentation like GitHub comments or Slack, as long as that conversation can be linked back to for future discussion. 

I think it's great you link back to the issue. In this case I don't think the comment makes the code any more readable -  someone six months from now may come back to this line of code and think "Why is this comment bringing "language" for this function?" 

My recommendation is to take this line of comment out, as it seems more of a discussion point rather than assisting in understanding the code for a future developer. But others may disagree

luisitocanlas committed:
Removed comment from line 494

luisitocanlas commented:
> > > @luisitocanlas I've reviewed this pretty thoroughly and think it looks good. I really appreciate your thorough PR description and the comments throughout.
> > > My only real question is related to L494, with the comment `//might need to add languages here, for issue #4648`. Could you clarify what this comment's purpose is?
> > 
> > 
> > @danvgar I figured since we're separating the technologies and languages, I added 'language' at the end of the searchAreas. But I didn't really see any effect when I added that. Do you want me to keep it or revert it back to what it was before? Or any other recommendations for that?
> > Thank you!
> 
> You're all good! I was mainly confused by the wording of it. My opinion is comments should be as declarative as possible, with any discrepancy or debate left for other forms of documentation like GitHub comments or Slack, as long as that conversation can be linked back to for future discussion.
> 
> I think it's great you link back to the issue. In this case I don't think the comment makes the code any more readable - someone six months from now may come back to this line of code and think "Why is this comment bringing "language" for this function?"
> 
> My recommendation is to take this line of comment out, as it seems more of a discussion point rather than assisting in understanding the code for a future developer. But others may disagree

Makes sense to me. I just remove the comment on line 494.

Thank you again!

danvgar reviewed (approved):
Thanks for that minor update, @luisitocanlas ! I was being nitpicky but this is a great job:

- Branch looks good
- Code was tested in docker and visually/functionally operates correctly. 
- PR description and comments are clear regarding issue explanation and purpose of changes

Nice work!

irais-valenzuela reviewed (approved):
@luisitocanlas I reviewed it again since you removed the comment on line 494. Everything looks good. Nice work!

colin-macrae reviewed (approved):
Great job, @luisitocanlas !
- The branch is going into the correct branch gh-pages. ✅
- The original issue [issue with #number] was properly referenced. ✅
- Per the instructions on the original issue, the specified lines of code were properly updated and functions correctly. ✅
